flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecLookupJoin.java

```diff
@@ -481,7 +516,14 @@ private StreamOperatorFactory<RowData> createAsyncLookupJoin(
                             isLeftOuterJoin,
                             asyncLookupOptions.asyncBufferCapacity);
         }
-
+        if (asyncLookupOptions.keyOrdered) {
+            checkNotNull(keySelector);
+            return new TableKeyedAsyncWaitOperatorFactory<>(
```

Comment:
I wonder if we need a new TableKeyedAsyncWaitOperatorFactory class, then there is the existing factory AsyncWaitOperatorFactory which seems the same apart from the keyselector parameter.